lib/features/manage_account/domain/usecases/edit_default_identity_interactor.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:core';
+
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/model/edit_default_identity_request.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/model/edit_identity_request.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/repository/identity_repository.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/state/edit_default_identity_state.dart';
+import 'package:tmail_ui_user/features/manage_account/presentation/profiles/identities/utils/identity_utils.dart';
+
+class EditDefaultIdentityInteractor {
+  final IdentityRepository _identityRepository;
+  final IdentityUtils _identityUtils;
+
+  EditDefaultIdentityInteractor(
+    this._identityRepository,
+    this._identityUtils
+  );
+
+  Stream<Either<Failure, Success>> execute(
+      AccountId accountId,
+      EditIdentityRequest editIdentityRequest
+  ) async* {
+    try {
+      yield Right(EditDefaultIdentityLoading());
+
+      final listDefaultIdentities = await _getDefaultIdentities(accountId);
+
+      // if the identity need to update in listDefaultIdentities, remove it
+      listDefaultIdentities
+          ?.removeWhere(
+              (identity) => identity.id == editIdentityRequest.identityId
+          );
```

Comment:
how to make it clear and short? one tip: create a method. Ex:
```suggestion
      listDefaultIdentities
          ?.removeWhere((identity) => requestIdentityStillDefault(editIdentityRequest.identityId, identityId));
          
          
       .......
       bool requestIdentityStillDefault(requestIdentity, defaultIdentityId) {
          //bla bla bla
       }
```